spec/std/db/dummy_driver.cr

```diff
@@ -0,0 +1,181 @@
+require "spec"
+
+class DummyDriver < DB::Driver
+  def build_connection(db : DB::Database) : DB::Connection
+    DummyConnection.new(db)
+  end
+
+  class DummyConnection < DB::Connection
+    def initialize(db)
+      super(db)
+      @@connections ||= [] of DummyConnection
+      @@connections.not_nil! << self
+    end
+
+    def self.connections
+      @@connections.not_nil!
+    end
+
+    def self.clear_connections
+      @@connections.try &.clear
+    end
+
+    def build_statement(query)
+      DummyStatement.new(self, query)
+    end
+
+    def last_insert_id : Int64
+      0
+    end
+
+    protected def do_close
+      super
+    end
+  end
+
+  class DummyStatement < DB::Statement
+    property params
+
+    def initialize(connection, @query)
+      @params = Hash(Int32 | String, DB::Any).new
+      super(connection)
+    end
+
+    protected def perform_query(args : Slice(DB::Any))
+      set_params args
+      DummyResultSet.new self, @query
+    end
+
+    protected def perform_exec(args : Slice(DB::Any))
+      set_params args
+      DB::ExecResult.new 0, 0
+    end
+
+    private def set_params(args)
+      @params.clear
+      args.each_with_index do |arg, index|
+        @params[index] = arg
+      end
+    end
+
+    protected def do_close
+      super
+    end
+  end
+
+  class DummyResultSet < DB::ResultSet
+    @@next_column_type = String
+
+    def initialize(statement, query)
+      super(statement)
+      @iterator = query.split.map { |r| r.split(',') }.to_a.each
+
+      @@last_result_set = self
+    end
+
+    protected def do_close
+      super
+    end
+
+    def self.last_result_set
+      @@last_result_set.not_nil!
+    end
+
+    def move_next
+      @iterator.next.tap do |n|
+        return false if n.is_a?(Iterator::Stop)
+        @values = n.each
+        return true
+      end
+    end
+
+    def column_count
+      2
+    end
+
+    def column_name(index)
+      "c#{index}"
+    end
+
+    def column_type(index : Int32)
+      @@next_column_type
+    end
+
+    def self.next_column_type=(value)
+      @@next_column_type = value
+    end
+
```

Comment:
I might miss something, but this looks as if a class var is used to transport non-global state?